checkov/json/runner.py

```diff
@@ -0,0 +1,73 @@
+import json
+import logging
+import os
+
+from checkov.common.output.record import Record
+from checkov.common.output.report import Report
+from checkov.common.runners.base_runner import BaseRunner, filter_ignored_paths
+from checkov.json.parser import parse
+from checkov.json.registry import registry
+from checkov.runner_filter import RunnerFilter
+
+
+class Runner(BaseRunner):
+    check_type = "json"
+
+    def run(self, root_folder=None, external_checks_dir=None, files=None, runner_filter=RunnerFilter(),
+            collect_skip_comments=True):
+        report = Report(self.check_type)
+        definitions = {}
+        definitions_raw = {}
+        parsing_errors = {}
+
+        def load_files(files_to_load, filename_fn=None, key_fn=None):
+            for file in files_to_load:
+                f = filename_fn(file) if filename_fn else file
+                key = key_fn(f) if key_fn else f
+                (definitions[key], definitions_raw[key]) = parse(f)
+
+        if not external_checks_dir:
+            logging.error("The json runner requires that external checks are defined.")
+            return report
+
+        for directory in external_checks_dir:
+            registry.load_external_checks(directory)
+
+        if files:
+            load_files(files)
+
+        if root_folder:
+            for root, d_names, f_names in os.walk(root_folder):
+                filter_ignored_paths(root, d_names, runner_filter.excluded_paths)
+                filter_ignored_paths(root, f_names, runner_filter.excluded_paths)
+                load_files(
+                    f_names,
+                    lambda f: os.path.join(root, f),
+                    # lambda k: f"{os.sep}{os.path.relpath(k, os.path.commonprefix((root_folder, k)))}"
+                )
+
+        for json_file_path in definitions.keys():
+            results = registry.scan(
+                json_file_path, definitions[json_file_path], [], runner_filter
+            )
+            for check, result in results.items():
+                result_config = result["results_configuration"]
+                start = result_config.start_mark.line
```

Comment:
are we populating this var with a value somewhere?

Reply:
Yes. This is using the json parser that is already in the codebase for terraform plans which captures line numbers along with the deserialized values. The custom checks should return the passing/failing configuration along with the check status. That configuration will have the start/end mark properties as a result of the tf plan parser. Does that make sense?

Reply:
I was also wondering if you would want to do something like move the tf plan parser into common as a general JSON parser? There doesn't seem to be anything terraform specific about it, but I didn't want to move something like that without discussing it first.

Reply:
i think it is the right thing to do. you'll need to rebase from master because we had a contribution with performance enhancement there a day ago